lib/index.js

```diff
@@ -146,16 +147,16 @@ Appboy.prototype.initializeV2 = function(customEndpoint) {
   };
 
   if (customEndpoint) config.baseUrl = customEndpoint;
-  
+
   this.initializeTester(options.apiKey, config);
   window.appboy.initialize(options.apiKey, config);
-  
-  if (options.automaticallyDisplayMessages) window.appboy.display.automaticallyShowNewInAppMessages();
-  if (userId) window.appboy.changeUser(userId);
-  
-  window.appboy.openSession();
 
-  this.load('v2', this.ready);
+  if (options.automaticallyDisplayMessages) window.appboy.display.automaticallyShowNewInAppMessages();
+  if (userId) {
+    window.appboy.changeUser(userId, window.appboy.openSession(self.load('v2', self.ready)));
```

Comment:
I think there's a minor oversight here.

`changeUser`'s second argument should be a function, no? Because the code as written, we'd be passing whatever `openSession` returns. I think we want to pass a function which in turn calls `openSession`.

A similar note for what we're passing to `openSession` -- we're passing `self.load(...)`, which isn't the right type.

In other words, I *think* we want to do:

```js
window.appboy.changeUser(userId, function () {
  window.appboy.openSession(function () {
    self.load('v2', self.ready);
  });
});
```

With the code as written, we'd evaluate `self.load(...)` as the *first* thing to happen, rather than the *last* thing.

Reply:
Yep -- thanks a bunch for catching this.